client/modules/IDE/components/Editor/index.jsx

```diff
@@ -65,564 +76,630 @@ import * as PreferencesActions from '../../actions/preferences';
+      beepRef.current = null;
     };
-    this._cm = null;
-    this.tidyCode = this.tidyCode.bind(this);
+  }, [beepUrl]);
 
-    this.updateLintingMessageAccessibility = debounce((annotations) => {
-      this.props.clearLintMessage();
+  const updateLintingMessageAccessibility = useCallback(
+    debounce((annotations) => {
+      clearLintMessage();
       annotations.forEach((x) => {
         if (x.from.line > -1) {
-          this.props.updateLintMessage(x.severity, x.from.line + 1, x.message);
+          updateLintMessage(x.severity, x.from.line + 1, x.message);
         }
       });
-      if (this.props.lintMessages.length > 0 && this.props.lintWarning) {
-        this.beep.play();
+      if (lintMessages.length > 0 && lintWarning) {
+        beepRef.current.play();
       }
-    }, 2000);
-    this.showFind = this.showFind.bind(this);
-    this.showReplace = this.showReplace.bind(this);
-    this.getContent = this.getContent.bind(this);
-  }
+    }, 2000),
+    []
+  );
 
-  componentDidMount() {
-    this.beep = new Audio(beepUrl);
-    // this.widgets = [];
-    this._cm = CodeMirror(this.codemirrorContainer, {
-      theme: `p5-${this.props.theme}`,
-      lineNumbers: this.props.lineNumbers,
-      styleActiveLine: true,
-      inputStyle: 'contenteditable',
-      lineWrapping: this.props.linewrap,
-      fixedGutter: false,
-      foldGutter: true,
-      foldOptions: { widget: '\u2026' },
-      gutters: ['CodeMirror-foldgutter', 'CodeMirror-lint-markers'],
-      keyMap: 'sublime',
-      highlightSelectionMatches: true, // highlight current search match
-      matchBrackets: true,
-      emmet: {
-        preview: ['html'],
-        markTagPairs: true,
-        autoRenameTags: true
-      },
-      autoCloseBrackets: this.props.autocloseBracketsQuotes,
-      styleSelectedText: true,
-      lint: {
-        onUpdateLinting: (annotations) => {
-          this.updateLintingMessageAccessibility(annotations);
-        },
-        options: {
-          asi: true,
-          eqeqeq: false,
-          '-W041': false,
-          esversion: 11
-        }
-      },
-      colorpicker: {
-        type: 'sketch',
-        mode: 'edit'
-      }
+  const editorRef = useRef(null);
+  const viewRef = useRef(null);
+  const pickrRef = useRef(null);
+  const fuseRef = useRef(null); // Store the Fuse instance in a ref
+  const docsRef = useRef({}); // Store the documents in a ref
+  const prevFileIdRef = useRef(null); // Store the previous file ID in a ref
+
+  useEffect(() => {
+    // Initialize the Fuse instance only when `hinter` changes
+    if (hinter && hinter.p5Hinter) {
+      fuseRef.current = new Fuse(hinter.p5Hinter, {
+        threshold: 0.05, // Set your threshold
+        keys: ['text'] // Keys to search against
+      });
+    }
+  }, [hinter]);
+
+  const [currentLine, setCurrentLine] = useState(1);
+
+  const replaceCommand = metaKey === 'Ctrl' ? `Mod-h` : `Mod-Alt-f`;
+
+  // Handle document changes (debounced)
```

Comment:
i don't see this being debounced? unless i missed something (very likely haha)? but if it's not being debounced lets change the comment or if it should be, let's add the debouncer

Reply:
yeah thanks for picking this up! I think i need to add back the debouncer. Will update it on the next commit!